metals/src/main/scala/scala/meta/internal/decorations/SyntheticsDecorationProvider.scala

```diff
@@ -157,6 +144,47 @@ final class SyntheticsDecorationProvider(
       pcHover
     }
 
+  private def syntheticDecorations(
+      path: AbsolutePath
+  ): Seq[DecorationOptions] = {
+
+    // If focused document is not supported we can't be sure what is currently open
+    def isFocusedDocument =
+      focusedDocument().contains(path) || !clientConfig.isDidFocusProvider()
+
+    /**
+     * Worksheets currently do not use semanticdb, which is why this will not work.
+     * If at any time worksheets will support it, we need to make sure that the
+     * evaluation will not be replaced with implicit decorations.
+     */
+    if (isFocusedDocument) {
+      val textDocument = currentDocument(path)
+      textDocument match {
+        case Some(doc) =>
+          decorations(path, doc)
+        case _ =>
+          Nil
+      }
+
+    } else {
+      Nil
+    }
```

Comment:
```suggestion
    currentDocument(path) match {
      case Some(doc) if isFocusedDocument =>
        decorations(path, doc)
      case _ =>
        Nil
    }
```
Mo Brackets, Mo Problems :D